### PR TITLE
dcc.Location props fix for callbacks

### PR DESCRIPTION
**Pull request by HammadTheOne**
As pointed out in #769, `dcc.Location` was firing callbacks for the `pathname` when the `href` was changed. This behaviour carried over to all props, including `hash` and `search`. This PR fixes that behaviour by adding in conditionals to check whether the new value is different from the previous one. 
 
To Do:

- [x] Add a basic integration test
- [x] Update changelog with PR tag and link.

### CODE REVIEW

**Comment by alexcjohnson** [comment]
Nice! supersedes community PR #765 - take a look at the app there as a test case.

File: src/components/Location.react.js [hunk 1]
```diff
@@ -106,7 +116,7 @@ export default class Location extends Component {
         this.updateLocation(this.props);
     }
 
-    UNSAFE_componentWillReceiveProps(nextProps) {
+    componentWillReceiveProps(nextProps) {
```
Comment (alexcjohnson) [inline src/components/Location.react.js]: We want to keep the `UNSAFE_` prefix - unless you can figure out how to use a different (safe) method.

Comment (HammadTheOne) [reply src/components/Location.react.js]: Hmm, that's strange, I don't even remember removing that. Might be a side-effect of formatting? 

In any case, re-added in c0e72b7. Thanks for pointing it out!

Comment (alexcjohnson) [reply src/components/Location.react.js]: I'd guess a mis-merge... or perhaps you had the file open already when you pulled the latest `dev`. Anyway it's fixed now, thanks!

**Comment by AHMatsWillemsen** [comment]
Hi, thanks for this! We found this bug while developing a multi-page app with one part listening for the pathname changes, and one part for the search-changes, resulting in double callbacks for the search callback. In what version do you expect this fix to be included? We can then remove a custom component.

**Comment by alexcjohnson** [comment]
@AHMatsWillemsen this fix is expected to be included in the next Dash version (1.10)

**Pull request merged**